src/client.cpp

```diff
@@ -393,7 +393,14 @@ status_t Client::Impl::uploadBackground(id_t accountId, id_t instrumentId,
 //    -F "data=@/path/to/image.png;type=image/png"    cs->addFormField(kStrKey, kStrFLIPBOOK);
     cs->addFormField(kStrKey, kStrBACKGROUND);
     cs->addFormField(kStrTimestamp, toIsoTimeString(timestamp));
-    cs->addFormFile(kStrData, imageFile, mimeTypeFromFilePath(imageFile));
+
+    if (payload.data)
+        cs->addFormFile(kStrData, payload.data, payload.dataSize, payload.mimeType.c_str());
```

Comment:
I also talked to Keith and he says that for frames with many objects json data for object stream becomes an issue on lower bandwidth connections.
I think for now we will stay with json, but for the future, we will need to use protobuf. We need web API to be ready for that though. So, now we do json.